### PR TITLE
Akismet Pro checkout line item prices should reflect license quantity

### PR DESCRIPTION
**Pull request by elliottprogrammer**
This PR Adjusts how prices are displayed in checkout for the Akismet Pro 500 product which has a _licensed quantity_, in order to make the price/total a little more clear to users.  (See before & after images:)

BEFORE | AFTER
--- | ---
Markup 2023-12-02 at 09 18 03 | Markup 2023-12-02 at 09 15 58

- Related discussion concerning dropdown price display: pbNhbs-8P6-p2#comment-18986
- PT: Akismet Checkout – Multiple licenses: pbNhbs-8My-p2
- Akismet Checkout: Multiple Licenses – Implementation Plan: pbNhbs-8P6-p2

## Testing Instructions

- (if not already merged), first apply patch D128326-code to your sandbox and sandbox `public-api.wordpress.com`.
- Checkout this PR branch and `yarn-start`
- Go to https://akismet.com/pricing and select/click the  "Get Pro" button. 
- On the checkout page, in the URL, replace `https://wordpress.com` with `http://calypso.localhost:3000` and press enter.
- You should see the "Number of licenses" dropdown, like in the image above.
- Select a different number of sites/licenses.
- Verify the way the prices are displayed make a little more sense.

**Note:** There is probably room for improvement here, and we probably need Design to iterate on the optimum price display, however for now I think we're just shooting for "acceptable" in order to get this project shipped and in front of our users asap (Targeting Monday Dec 4th).

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/checkout/src/components/akismet-pro-quantity-dropdown/index.tsx [hunk 1]
```diff
@@ -104,6 +105,22 @@ const OptionList = styled.ul`
 	}
 `;
 
+const CurrentOptionContainer = styled.div`
+	align-items: center;
+	display: flex;
+	font-size: 14px;
```
Comment (CodeyGuyDylan) [inline client/my-sites/checkout/src/components/akismet-pro-quantity-dropdown/index.tsx]: Looks like you can use `font-size: ${ ( props ) => props.theme.fontSize.small };` here, I found the values in this file: fbhepr%2Skers%2Spnylcfb%2Scnpxntrf%2Spbzcbfvgr%2Qpurpxbhg%2Sfep%2Syvo%2Sgurzr.gf%3Se%3Q4o0qo534%23110-og

Additionally below you can use `font-weight: ${ ( props ) => props.theme.weights.normal };`

File: client/my-sites/checkout/src/components/akismet-pro-quantity-dropdown/index.tsx [hunk 2]
```diff
@@ -271,6 +288,33 @@ export const AkismetProQuantityDropDown: FunctionComponent< AkismetProQuantityDr
...
+		stripZeros: true,
+	} );
+
+	const actualAmountDisplay = formatCurrency( item_subtotal_integer, currency, {
+		isSmallestUnit: true,
+		stripZeros: true,
+	} );
+
+	const currentOptionPriceDisplay = translate(
+		'%(quantity)d licenses X %(actualAmountQuantityOneDisplay)s per license = %(actualAmountDisplay)s',
```
Comment (CodeyGuyDylan) [inline client/my-sites/checkout/src/components/akismet-pro-quantity-dropdown/index.tsx]: Do we need a translators comment for this string?

File: packages/wpcom-checkout/src/checkout-line-items.tsx [hunk 3]
```diff
@@ -1069,20 +1071,26 @@ function CheckoutLineItem( {
...
 	const label = getLabel( product );
+	const itemOriginalSubtotalInteger =
+		isAkPro500Cart && product.quantity
+			? product.item_original_cost_for_quantity_one_integer
+			: product.item_original_subtotal_integer;
+	const originalAmountDisplay = formatCurrency( itemOriginalSubtotalInteger, product.currency, {
+		isSmallestUnit: true,
+		stripZeros: true,
+	} );
+	const originalAmountInteger = itemOriginalSubtotalInteger;
```
Comment (CodeyGuyDylan) [inline packages/wpcom-checkout/src/checkout-line-items.tsx]: I don't think we need this variable anymore if it is always equal to `itemOriginalSubtotalInteger`, and seeing as they are both defined as constants, they should always be equal

**Review by CodeyGuyDylan — CHANGES_REQUESTED** [review]
I am unsure if this is actually a problem, but I noticed that if you switch to Akismet Business via the dropdown, and refresh the page, the dropdown disappears and we can no longer switch back to Pro with 1-4 sites. I think I understand that we want to hide it if the user clicks on the Business card, I am just wondering if that could be confusing to somebody after refresh being unable to switch back to what they were at before? I am not sure. Anyway here is a screen recording of what I am talking about


https://github.com/Automattic/wp-calypso/assets/65001528/b9075097-c457-4684-af01-050e19264f32

Additionally the Akismet Business plan looks a bit weird with the line wrapping

image

Is this possible to avoid? I could see it getting pretty cramped so I am not sure if it is. If it's unavoidable, and you haven't already, I'd reach out to design to make sure we make it look as good as possible

Additionally none of them look very good on mobile 🤔 

image

Other than a few nitpick comments the code looks good

**Comment by jboland88** [comment]
> I am unsure if this is actually a problem, but I noticed that if you switch to Akismet Business via the dropdown, and refresh the page, the dropdown disappears and we can no longer switch back to Pro with 1-4 sites. I think I understand that we want to hide it if the user clicks on the Business card, I am just wondering if that could be confusing to somebody after refresh being unable to switch back to what they were at before?

@CodeyGuyDylan - does this only happen when you refresh the page? If so, I think this behavior is OK. I think it would be unusual for someone to refresh the checkout page in the middle of the checkout process?

**Comment by CodeyGuyDylan** [comment]
> > I am unsure if this is actually a problem, but I noticed that if you switch to Akismet Business via the dropdown, and refresh the page, the dropdown disappears and we can no longer switch back to Pro with 1-4 sites. I think I understand that we want to hide it if the user clicks on the Business card, I am just wondering if that could be confusing to somebody after refresh being unable to switch back to what they were at before?
> 
> @CodeyGuyDylan - does this only happen when you refresh the page? If so, I think this behavior is OK. I think it would be unusual for someone to refresh the checkout page in the middle of the checkout process?

Yes only when the page is refreshed

**Comment by elliottprogrammer** [comment]
> > > I am unsure if this is actually a problem, but I noticed that if you switch to Akismet Business via the dropdown, and refresh the page, the dropdown disappears and we can no longer switch back to Pro with 1-4 sites. I think I understand that we want to hide it if the user clicks on the Business card, I am just wondering if that could be confusing to somebody after refresh being unable to switch back to what they were at before?
> > 
> > 
> > @CodeyGuyDylan - does this only happen when you refresh the page? If so, I think this behavior is OK. I think it would be unusual for someone to refresh the checkout page in the middle of the checkout process?
> 
> Yes only when the page is refreshed

@CodeyGuyDylan, @jboland88 - Yes, I actually did this on purpose.  If I did it otherwise, any time a user directly chooses Akismet Businnes (from Akismet.com) and goes directly to the cart, then the Akismet Pro quantity dropdown will appear and allow them (possibly encourage them) to downgrade to an Akismet Pro plan instead.  With the way it is, the option to select Akismet Business using the quantity dropdown selector is only available when the user has first decided to checkout with Akismet Pro first. 
Yes, as a consequence, if they refresh the page when Akismet Business is in the cart then they will no longer see the dropdown to downgrade, but I think it's a fair trade-off and better than the alternative.

File: client/my-sites/checkout/src/components/akismet-pro-quantity-dropdown/index.tsx [hunk 4]
```diff
@@ -271,6 +288,33 @@ export const AkismetProQuantityDropDown: FunctionComponent< AkismetProQuantityDr
...
+		{
+			args: {
+				quantity: quantity ?? 1,
+				actualAmountQuantityOneDisplay,
+				actualAmountDisplay,
+			},
+			components: {
+				s: <s />,
```
Comment (sirbrillig) [inline client/my-sites/checkout/src/components/akismet-pro-quantity-dropdown/index.tsx]: 🤔 Is this tag used in the string somewhere? I don't see it.

File: packages/wpcom-checkout/src/checkout-line-items.tsx (hunk 3 above)
Comment (elliottprogrammer) [reply packages/wpcom-checkout/src/checkout-line-items.tsx]: Yes, thats for catching that. 👍

File: client/my-sites/checkout/src/components/akismet-pro-quantity-dropdown/index.tsx (hunk 4 above)
Comment (elliottprogrammer) [reply client/my-sites/checkout/src/components/akismet-pro-quantity-dropdown/index.tsx]: Oops, I forgot to take that out. Thanks for catching that! 👍

**Review by CodeyGuyDylan — CHANGES_REQUESTED** [review]
I noticed when switching to the Akismet Business plan, it looks bad visually for a second

https://github.com/Automattic/wp-calypso/assets/65001528/3403f93d-3d4e-4dd3-ad40-bb05dd37e102

I know it's a small thing but if this is possible to avoid I think we should 🤔 

Other than that, all the changes look good!

**Comment by elliottprogrammer** [comment]
> I noticed when switching to the Akismet Business plan, it looks bad visually for a second
> 
> I know it's a small thing but if this is possible to avoid I think we should 🤔
> Other than that, all the changes look good!

Ahh yes agreed, that doesn't look so good. I can fix that, I'm pretty sure.

Ok @CodeyGuyDylan, how about now? That seems better, right?

File: client/my-sites/checkout/src/components/akismet-pro-quantity-dropdown/index.tsx (hunk 2 above)
Comment (elliottprogrammer) [reply client/my-sites/checkout/src/components/akismet-pro-quantity-dropdown/index.tsx]: Yes, good idea, wouldn't hurt! 👍

**Review by CodeyGuyDylan — APPROVED** [review]
Yes that looks much better! Thank you for updating that, LGTM

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10069758

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @elliottprogrammer for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10069758).

**Pull request merged**